src/pip/_internal/utils/filesystem.py

```diff
@@ -180,3 +180,12 @@ def directory_size(path: str) -> Union[int, float]:
 
 def format_directory_size(path: str) -> str:
     return format_size(directory_size(path))
+
+
+def list_empty_subdirs(path):
+    # type: (str) -> List[str]
+    """Returns a list of absolute paths of empty directories beneath path."""
+    result = []  # type: List[str]
+    for root, dirs, _files in os.walk(path, topdown=False):
+        result.extend(os.path.join(root, d) for d in dirs)
```

Comment:
We can `yield from ...` here and avoid creating a list.

Reply:
In the process of trying to figure out how to use `yield from ...` here, I found a bug in `list_empty_subdirs()`. The bug is it doesn't even check if the directory is empty. :joy:

Reply:
🤣